File: ios/Plugin/TreadsGps3Plugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(TreadsGps3Plugin, "TreadsGps3",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(keepWebviewAwake, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(checkStatus, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestIgnoreBatteryOptimization, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestIgnoreDataSaver, CAPPluginReturnPromise);
)

File: ios/Plugin/TreadsGps3Plugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(TreadsGps3Plugin)
public class TreadsGps3Plugin: CAPPlugin {
    private let implementation = TreadsGps3()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func keepWebviewAwake(_ call: CAPPluginCall) {
      call.resolve([
          "success": true
      ])
    }

    @objc func checkStatus(_ call: CAPPluginCall) {
      call.resolve([
          "dataSaver": "unknown",
          "ignoreBatteryOptimization": "unknown"
      ])
    }

    @objc func requestIgnoreBatteryOptimization(_ call: CAPPluginCall) {
      call.resolve([
          "success": false
      ])
    }

    @objc func requestIgnoreDataSaver(_ call: CAPPluginCall) {
      call.resolve([
          "success": false
      ])
    }
}
